src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/EntrypointSelectionStep.cs

```diff
@@ -90,57 +96,36 @@ protected UpgradeStepInitializeResult InitializeImpl(IUpgradeContext context)
             if (!context.InputIsSolution)
             {
                 var project = projects.First(i => i.FileInfo.Name.Equals(Path.GetFileName(context.InputPath), StringComparison.OrdinalIgnoreCase));
-                context.SetEntryPoint(project);
+                context.EntryPoints = new[] { project };
 
                 Logger.LogInformation("Setting entrypoint to user selected project: {Project}", project.FileInfo);
 
                 return new UpgradeStepInitializeResult(UpgradeStepStatus.Complete, "Selected user's choice of entry point project.", BuildBreakRisk.None);
             }
 
-            if (!string.IsNullOrEmpty(_entryPoint))
+            context.EntryPoints = _entrypointResolver.GetEntrypoints(context.Projects, _options.EntryPoint);
+
+            if (context.EntryPoints.Any())
             {
-                var entryPointProject = projects.FirstOrDefault(i => i.FileInfo.Name.Equals(_entryPoint, StringComparison.OrdinalIgnoreCase));
-                if (entryPointProject is not null)
-                {
-                    context.SetEntryPoint(entryPointProject);
-                    Logger.LogInformation("Setting entrypoint to user selected project in solution: {Project}", _entryPoint);
-
-                    return new UpgradeStepInitializeResult(UpgradeStepStatus.Complete, "Selected user's choice of entry point project.", BuildBreakRisk.None);
-                }
-                else
-                {
-                    Logger.LogInformation("Invalid Entry-Point Project specified : {Project}", _entryPoint);
-                }
+                return new UpgradeStepInitializeResult(UpgradeStepStatus.Complete, "Selected user's choice of entry point project.", BuildBreakRisk.None);
             }
-
-#pragma warning disable CA1508 // Avoid dead conditional code
-            if (context.EntryPoint is null && _isNonInteractiveMode)
-#pragma warning restore CA1508 // Avoid dead conditional code
+            else
             {
-                throw new UpgradeException("Entry Point needs to be provided when more than 1 projects present in a non-interative mode. There are 2 ways of providing an entry-point :\n" +
-                    "a) Execute upgrade-assistant in interactive mode up until select entry point step and re-run upgrade-assistant in non-interactive mode \n" +
-                    "b) Execute upgrade-assistant in non-interactive mode with valid value for --entry-point");
+                return new UpgradeStepInitializeResult(UpgradeStepStatus.Failed, "No entrypoint was selected. Solutions require an entrypoint to proceed.", BuildBreakRisk.None);
```

Comment:
Shouldn't we return Incomplete in this scenario so that the user can execute this step to select an entry point? As this change reads now, `EntrypointSelectionStep`'s `InitializeImplAsync` never returns incomplete - only complete or failed, so the user *has* to specify -e. They won't get an option of choosing an entrypoint interactively. Is that intentional?

Reply:
Yup!

Reply:
This line should retain the exception message from before.

Reply:
Reverted that